src/mechanics/cell/Cell.h

```diff
@@ -35,14 +36,14 @@ class Cell : public CellInterface
 
     void Apply(VoidFunction f) override
     {
-        CellData cellData(mElements);
+        CellData cellData(mElements, mIntegrationType.GetNumIntegrationPoints(), Id());
         for (int iIP = 0; iIP < mIntegrationType.GetNumIntegrationPoints(); ++iIP)
         {
             auto ipCoords = mIntegrationType.GetLocalIntegrationPointCoordinates(iIP);
             Jacobian jacobian(mElements.CoordinateElement().ExtractNodeValues(),
                               mElements.CoordinateElement().GetDerivativeShapeFunctions(ipCoords),
                               mElements.CoordinateElement().GetDofDimension());
-            CellIpData cellipData(mElements, jacobian, ipCoords);
+            CellIpData cellipData(mElements, jacobian, ipCoords, iIP);
```

Comment:
I vote for keeping the ids separate. This allows data at the integrand that is available once per _cell_, not once per _ip_. I can think of the cell area in a regularized local damage model. The integrand would have two data members:
- `mHistoryData` - `numCells x numIpPerCell` (access via cellId _and_ ipId)
- `mCellArea` - `numCells` (access _only_ via cellId)

Reply:
I also think, providing a single number is kind of an unwanted constraint. The user now has to use a data type which takes a single number as key.

Reply:
IMO should the integrand not be aware of the underlying cell structure, the area e.g. is no history data, but rather cell data (that could be recalculated each time from the CellData). The user just provides an interface for his integrand with a certain number of history data - that's all the standard integrand writing user should deal with.

Reply:
Well if we want to hide the ID from the user (he can't set an ID himself) this discussion might be relevant. If the user can set the cell ID himself (via ctor), he can decide which approach he favors ("ID" or start in container) by picking his IDs accordingly. Since he his also able to use his own history data approach he can do whatever he likes.

Reply:
@joergfunger What is the benefit of this single id? The only one I see is not storing an `int` in the `CellData` object. But this is no problem.
Everything else, like passing the two ids to a law, is somehow part of the integrand. @vhirtham chose for his law to get two ids. We can design a default momentum balance integrand that calculates a single continuous id and passes that to the `MechanicsInterface`. 
But I see no reason in limiting _all_ integrands to that.